### PR TITLE
Mandate minimum TLS version of 1.2 for all storage accounts

### PR DESCRIPTION
**Pull request by craddm**
## :white_check_mark: Checklist

- [x] You have given your pull request a meaningful title (_e.g._ `Enable foobar integration` rather than `515 foobar`).
- [x] You are targeting the appropriate branch. If you're not certain which one this is, it should be **`develop`**.
- [ ] Your branch is up-to-date with the **target branch** (it probably was when you started, but it may have changed since then).

### :arrow_heading_up: Summary

As per pentest report, storage accounts in SREs currently allow a minimum TLS version of 1.0, which is deprecated.

Note that this also applies to the SHM storage account, and this PR also modifies the minimum version there.

### :microscope: Tests

Deployed SRE using these changes.

- [x] mounting the accounts inside the SRE
- [x] reading/writing from inside the SRE
- [x] accessing them from Azure Store Explorer

### CODE REVIEW

**Comment by jemrobinson** [comment]
Is this a from-scratch SRE? I'd like to know whether this causes any problems for any of:

- mounting the accounts inside the SRE
- reading/writing (as appropriate) from inside the SRE
- accessing them from Azure Storage Explorer

Could you give details on this issue when you have them?

**Comment by craddm** [comment]
> Is this a from-scratch SRE? I'd like to know whether this causes any problems for any of:
> 
> * mounting the accounts inside the SRE
> * reading/writing (as appropriate) from inside the SRE
> * accessing them from Azure Storage Explorer
> 
> Could you give details on this issue when you have them?

It isn't a from scratch SRE - I deleted and redeployed the storage accounts. However, currently having issues restarting the `gitea` container group. Going to try a fresh deployment instead.

**Comment by craddm** [comment]
> Is this a from-scratch SRE? I'd like to know whether this causes any problems for any of:
> 
> * mounting the accounts inside the SRE
> * reading/writing (as appropriate) from inside the SRE
> * accessing them from Azure Storage Explorer
> 
> Could you give details on this issue when you have them?

All worked fine as far as I can tell. Successfully uploaded and downloaded from ingress and egress via storage explore. Successfully copied from input and to output from the SRE desktop

**Comment by jemrobinson** [comment]
Cool - that's /ingress and /egress. Can you check /home (presumably working if you logged in) and /shared as well?

File: data_safe_haven/external/api/azure_sdk.py [hunk 1]
```diff
@@ -581,6 +581,7 @@ def ensure_storage_account(
                     kind=StorageAccountKind.STORAGE_V2,
                     location=location,
                     tags=tags,
+                    minimum_tls_version="TLS1_2",
```
Comment (JimMadge) [inline data_safe_haven/external/api/azure_sdk.py]: Is there an enum from the Azure SDK we can use here instead of a raw string?

**Review by JimMadge — COMMENTED** [review]
LGTM. We should be careful to test this thoroughly though as we are merging to the release branch.

File: data_safe_haven/external/api/azure_sdk.py (hunk 1 above)
Comment (jemrobinson) [reply data_safe_haven/external/api/azure_sdk.py]: `storage.MinimumTlsVersion`

Comment (JimMadge) [reply data_safe_haven/external/api/azure_sdk.py]: Is that `pulumi.azure_native`? I think this is the Python Azure SDK.

Comment (jemrobinson) [reply data_safe_haven/external/api/azure_sdk.py]: Sorry - you're right!

Comment (jemrobinson) [reply data_safe_haven/external/api/azure_sdk.py]: It's from `azure.mgmt.storage.v2021_08_01.models import MinimumTlsVersion`

Comment (craddm) [reply data_safe_haven/external/api/azure_sdk.py]: done

**Comment by jemrobinson** [comment]
@JimMadge : I'm comfortable that Matt has tested /data and /output (also /home and maybe /shared). Do you think we need/want an independent test of these?

**Comment by craddm** [comment]
> @JimMadge : I'm comfortable that Matt has tested /data and /output (also /home and maybe /shared). Do you think we need/want an independent test of these?

Yes, home and shared working fine too 

Just now deployed an SHM with the updated TLS too

**Comment by jemrobinson** [comment]
Happy to merge this @JimMadge ?

**Review by JimMadge — APPROVED** [review]

**Pull request merged**